xls/contrib/mlir/IR/xls_ops.cc

```diff
@@ -620,6 +620,11 @@ ParseResult SchanOp::parse(OpAsmParser& parser, OperationState& result) {
   result.addAttribute("type", TypeAttr::get(type));
   result.types.push_back(SchanType::get(parser.getContext(), type, false));
   result.types.push_back(SchanType::get(parser.getContext(), type, true));
+
+  if (parser.parseOptionalAttrDictWithKeyword(result.attributes)) {
```

Comment:
One could also do

return failure(parser.parseOptionalAttrDictWithKeyword(result.attributes));

to collapse conditional.